app/packages/core/src/components/Sidebar/Entries/FilterEntry.tsx

```diff
@@ -32,6 +32,8 @@ const Filter = ({ modal }: { modal: boolean }) => {
   );
   const affectedPathCount = useRecoilValue(affectedPathCountState);
 
+  const { setSelectedFieldsStage } = fos.useSchemaSettings();
```

Comment:
To clear the selection when eyeVisibility clicked
<img width="300" alt="Screen Shot 2023-05-10 at 3 02 28 PM" src="https://github.com/voxel51/fiftyone/assets/109545780/331bd06f-96fb-4069-87fa-7888acb40c2a">